src/sdam/public.rs

```diff
@@ -133,3 +173,125 @@ impl<'a> fmt::Display for ServerInfo<'a> {
         write!(f, " }}")
     }
 }
+
+/// A description of the most up-to-date information known about a topology. Further details can
+/// be found in the [Server Discovery and Monitoring specification](https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst).
+#[derive(Clone)]
+pub struct TopologyInfo {
+    pub(crate) description: TopologyDescription,
+}
+
+impl From<TopologyDescription> for TopologyInfo {
+    fn from(description: TopologyDescription) -> Self {
+        Self { description }
+    }
+}
+
+impl TopologyInfo {
+    /// Whether this topology has a readable server available that satisfies the specified selection
+    /// criteria.
+    pub fn has_readable_server(&self, selection_criteria: Option<SelectionCriteria>) -> bool {
+        match self.description.suitable_servers_in_latency_window(
+            &selection_criteria
+                .unwrap_or(SelectionCriteria::ReadPreference(ReadPreference::Primary)),
+        ) {
+            Ok(servers) => !servers.is_empty(),
+            Err(_) => false,
+        }
+    }
+
+    /// Whether this topology has a writable server available.
+    pub fn has_writable_server(&self) -> bool {
+        match self.description.topology_type {
+            TopologyType::Unknown | TopologyType::ReplicaSetNoPrimary => false,
+            TopologyType::Single | TopologyType::Sharded => {
+                self.description.has_available_servers()
+            }
+            TopologyType::ReplicaSetWithPrimary => true,
+        }
+    }
+
+    /// Gets the type of the topology.
```

Comment:
Added these getters for the same reasoning described [here](https://github.com/mongodb/mongo-rust-driver/pull/404#discussion_r677802805)